hubspot_sync/api.py

```diff
@@ -433,53 +615,70 @@ def sync_deal_with_hubspot(order_id: int) -> SimplePublicObject:
     )
 
     for line in order.lines.all():
-        sync_line_item_with_hubspot(line.id)
+        sync_line_item_with_hubspot(line)
     return result
 
 
-def sync_product_with_hubspot(product_id: int) -> SimplePublicObject:
+def sync_product_with_hubspot(product: Product) -> SimplePublicObject:
     """
     Sync a Product with a hubspot product
 
     Args:
-        product_id(int): The Product id
+        product(Product): The Product object.
 
     Returns:
         SimplePublicObject: The hubspot product object
     """
-    body = make_product_sync_message(product_id)
+    body = make_product_sync_message_from_product(product)
     content_type = ContentType.objects.get_for_model(Product)
 
-    # Check if a matching hubspot object has been or can be synced
-    get_hubspot_id_for_object(Product.objects.get(id=product_id))
-
     return upsert_object_request(
-        content_type, HubspotObjectType.PRODUCTS.value, object_id=product_id, body=body
+        content_type, HubspotObjectType.PRODUCTS.value, object_id=product.id, body=body
     )
 
 
-def sync_contact_with_hubspot(user_id: int) -> SimplePublicObject:
+def sync_contact_with_hubspot(user: User):
     """
-    Sync a user with a hubspot_sync contact
+    Sync a list of User objects with their hubspot_sync contacts.
 
     Args:
-        user_id(int): The User id
+        user User: User object.
 
     Returns:
-        SimplePublicObject: The hubspot contact object
+        bool: True if the contact upsert to HubSpot was successful, otherwise False.
+
+    Raises:
+        ApiException: Raised if HubSpot upsert request fails.
     """
-    body = make_contact_sync_message(user_id)
     content_type = ContentType.objects.get_for_model(User)
-    result = upsert_object_request(
-        content_type, HubspotObjectType.CONTACTS.value, object_id=user_id, body=body
-    )
-    User.objects.filter(id=user_id).update(hubspot_sync_datetime=now_in_utc())
-    return result
+    body = make_contact_sync_message_from_user(user)
+    try:
+        upsert_object_request(
+            content_type,
+            HubspotObjectType.CONTACTS.value,
+            object_id=user.id,
+            body=body,
+        )
+    except ApiException:
+        return False
+    time.sleep(settings.HUBSPOT_TASK_DELAY / 1000)
+
+    user.hubspot_sync_datetime = now_in_utc()
+    user.save()
+
+    return True
```

Comment:
I wasn't sure what I would return if the exception is thrown.  I figured returning a boolean was pretty straightforward and I didn't see any place where we were interpreting the result.

Reply:
Maybe best to change them all to return True or nothing at all, for consistency, since the return value isn't used anyway.  And allow any exception thrown by `upsert_object_request ` to be raised so we're aware of potential problems.